lua/tardis-nvim/init.lua

```diff
@@ -47,26 +49,37 @@ local function force_delete_buffer(buffer)
     return function() vim.api.nvim_buf_delete(buffer, { force = true }) end
 end
 
-local function commit_message(commit)
+local function build_commit_buffer_name(commit, buffer)
+    return string.format('%s#%s (%s)', constants.name_prefix, commit, buffer)
+end
+
+local function commit_message(commit, root)
     return function ()
         local message = Job:new({
             command = 'git',
-            args = { '-C', get_git_root(), 'show', '--compact-summary', commit }
+            args = { '-C', root, 'show', '--compact-summary', commit }
         }):sync()
 
         local buffer = vim.api.nvim_create_buf(false, true)
         vim.keymap.set('n', config.keymap.quit, force_delete_buffer(0), { buffer = buffer })
         vim.api.nvim_buf_set_lines(buffer, 0, -1, false, message)
         vim.api.nvim_buf_set_option(buffer, 'filetype', 'gitcommit')
         vim.api.nvim_buf_set_option(buffer, 'readonly', true)
-        vim.api.nvim_buf_set_name(buffer, constants.name_prefix .. 'message')
+        vim.api.nvim_buf_set_name(buffer, build_commit_buffer_name(commit, buffer))
```

Comment:
I changed the names of the buffers a little bit to make them unique. I also think it looks a little nicer.